Include/cpython/pyatomic_gcc.h

```diff
@@ -0,0 +1,679 @@
+#ifndef Py_ATOMIC_GCC_H
+#  error "this header file must not be included directly"
+#endif
+
+// This is the implementation of Python atomic operations using GCC's built-in
+// functions that match the C+11 memory model. This implementation is preferred
+// for GCC compatible compilers, such as Clang. These functions are available in
+// GCC 4.8+ without needing to compile with --std=c11 or --std=gnu11.
+
+static inline int
+_Py_atomic_add_int(int *address, int value)
+{
+    return __atomic_fetch_add(address, value, __ATOMIC_SEQ_CST);
+}
+
+static inline int8_t
+_Py_atomic_add_int8(int8_t *address, int8_t value)
+{
+    return __atomic_fetch_add(address, value, __ATOMIC_SEQ_CST);
+}
+
+static inline int16_t
+_Py_atomic_add_int16(int16_t *address, int16_t value)
+{
+    return __atomic_fetch_add(address, value, __ATOMIC_SEQ_CST);
+}
+
+static inline int32_t
+_Py_atomic_add_int32(int32_t *address, int32_t value)
+{
+    return __atomic_fetch_add(address, value, __ATOMIC_SEQ_CST);
+}
+
+static inline int64_t
+_Py_atomic_add_int64(int64_t *address, int64_t value)
+{
+    return __atomic_fetch_add(address, value, __ATOMIC_SEQ_CST);
+}
+
+static inline intptr_t
+_Py_atomic_add_intptr(intptr_t *address, intptr_t value)
+{
+    return __atomic_fetch_add(address, value, __ATOMIC_SEQ_CST);
+}
+
+static inline unsigned int
+_Py_atomic_add_uint(unsigned int *address, unsigned int value)
+{
+    return __atomic_fetch_add(address, value, __ATOMIC_SEQ_CST);
+}
+
+static inline uint8_t
+_Py_atomic_add_uint8(uint8_t *address, uint8_t value)
+{
+    return __atomic_fetch_add(address, value, __ATOMIC_SEQ_CST);
+}
+
+static inline uint16_t
+_Py_atomic_add_uint16(uint16_t *address, uint16_t value)
+{
+    return __atomic_fetch_add(address, value, __ATOMIC_SEQ_CST);
+}
+
+static inline uint32_t
+_Py_atomic_add_uint32(uint32_t *address, uint32_t value)
+{
+    return __atomic_fetch_add(address, value, __ATOMIC_SEQ_CST);
+}
+
+static inline uint64_t
+_Py_atomic_add_uint64(uint64_t *address, uint64_t value)
+{
+    return __atomic_fetch_add(address, value, __ATOMIC_SEQ_CST);
+}
+
+static inline uintptr_t
+_Py_atomic_add_uintptr(uintptr_t *address, uintptr_t value)
+{
+    return __atomic_fetch_add(address, value, __ATOMIC_SEQ_CST);
+}
+
+static inline Py_ssize_t
+_Py_atomic_add_ssize(Py_ssize_t *address, Py_ssize_t value)
+{
+    return __atomic_fetch_add(address, value, __ATOMIC_SEQ_CST);
+}
+
+
+static inline int
+_Py_atomic_compare_exchange_int(int *address, int expected, int value)
+{
+    return __atomic_compare_exchange_n(address, &expected, value, 0, __ATOMIC_SEQ_CST, __ATOMIC_SEQ_CST);
```

Comment:
Can you try to git into 80 columns? (Write this call in two lines.) Same remark for code below.